tldw_Server_API/app/core/DB_Management/ManuscriptDB.py

```
@@ -0,0 +1,824 @@
+            else:
+                set_parts.append(f"{key} = ?")
+                params.append(value)
+
+        set_parts.extend(["last_modified = ?", "version = ?", "client_id = ?"])
+        params.extend([now, next_version, self._client_id])
+        params.extend([project_id, expected_version])
+
+        with self.db.transaction() as conn:
+            cur = conn.execute(
+                f"UPDATE manuscript_projects SET {', '.join(set_parts)} "  # nosec B608
+                "WHERE id = ? AND version = ? AND deleted = 0",
```

Ruff (0.15.9)

[error] Possible SQL injection vector through string-based query construction

(S608)